src/carousel.tsx

```diff
@@ -0,0 +1,693 @@
+                carouselEl.scrollTo({left: scroll, behavior: 'smooth'});
+            }
+        },
+        [pagesScrollPositions]
+    );
+
+    const goPrev = React.useCallback(() => {
+        const carouselEl = carouselRef.current;
+        if (carouselEl) {
+            const {scrollLeft} = carouselEl;
+            const prevPageScrollPosition = [...scrollPositions]
+                .reverse()
+                .find((pos) => pos - scrollLeft < -1);
+            console.log('scrollLeft', scrollLeft, 'prevScrollPosition', prevPageScrollPosition);
+            carouselEl.scrollTo({left: prevPageScrollPosition, behavior: 'smooth'});
+        }
+    }, [scrollPositions]);
+
+    const goNext = React.useCallback(() => {
+        const carouselEl = carouselRef.current;
+        if (carouselEl) {
+            const {scrollLeft} = carouselEl;
+            const nextPageScrollPosition = scrollPositions.find((pos) => pos - scrollLeft > 1);
+            console.log('scrollLeft', scrollLeft, 'nextPageScrollPosition', nextPageScrollPosition);
+            carouselEl.scrollTo({left: nextPageScrollPosition, behavior: 'smooth'});
+        }
+    }, [scrollPositions]);
+
+    React.useEffect(() => {
+        if (autoplay) {
+            const time = typeof autoplay === 'boolean' ? 5000 : autoplay.time;
+            const interval = setInterval(() => {
+                if (scrollRight !== 0) {
+                    goNext();
+                }
+            }, time);
+            return () => clearInterval(interval);
+        }
+    }, [autoplay, goNext, scrollRight]);
+
+    const currentPageIndex = calcCurrentPageIndex(scrollLeft, pagesScrollPositions);
+
+    React.useEffect(() => {
+        if (onPageChange) {
+            const lastShownItemIndex = Math.min(
+                (currentPageIndex + 1) * itemsPerPageFloor - 1,
+                items.length - 1
+            );
+            const shownItemIndexes = [];
+            for (let i = 0; i < itemsPerPageFloor; i++) {
+                shownItemIndexes.unshift(lastShownItemIndex - i);
+            }
+            onPageChange({pageIndex: currentPageIndex, shownItemIndexes});
+        }
+    }, [currentPageIndex, items.length, itemsPerPageFloor, onPageChange]);
+
+    let bullets: React.ReactNode = null;
+
+    if (renderBullets) {
+        bullets = renderBullets({numPages: pagesCount, currentIndex: currentPageIndex, onPress: goToPage});
+    } else if (withBullets) {
+        bullets = <PageBullets numPages={pagesCount} currentIndex={currentPageIndex} onPress={goToPage} />;
+    }
+
+    return (
+        <Stack space={24}>
+            <div className={classes.carouselContainer}>
+                <Touchable
+                    className={classNames(classes.arrowButton, 'prev')}
+                    aria-label={texts.carouselPrevButton}
+                    onPress={goPrev}
+                    disabled={!showPrevArrow}
+                >
+                    <IconChevronLeftRegular />
```

Comment:
When you use carousel inside an inverse context with ThemeVariant, arrows disappear
https://tinyurl.com/y8yu3v9k

![image](https://user-images.githubusercontent.com/6722153/152359348-169ea2d9-85b0-44a1-b25b-15f735ef1627.png)